contracts/Removal.sol

```diff
@@ -688,6 +750,12 @@ contract Removal is
     }
   }
 
+  /**
+   * @dev Validates that the provided `id` should be minted.
```

Comment:
I feel like we might be overly relying on the `@dev` tags here.
Generally, if we're keeping documentation to a minimum, I'd prefer we rely on `@notice` over `@dev`. [From the natspec docs](https://docs.soliditylang.org/en/latest/natspec-format.html):

> @notice - Explain to an end user what this does
> @dev - Explain to a developer any extra details

So `@notice` is meant to help a reader understand what a function does as a baseline, and then @dev is where you put any extra implementation details to help a developer understand how it works. I think generally it's preferable to have some kind of `@notice` to explain what something does at a high level to a general audience, and then optionally have `@dev` flags to explain any subtle implementation details that a developer should know.

I don't know what Jaycen meant by the todo " @dev vs @notice consistency", but I think the idea behind being consistent with `@dev` and `@notice`.